server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsRequest.java

```diff
@@ -80,7 +86,11 @@ public GetSnapshotsRequest(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeString(repository);
+        if (out.getVersion().onOrAfter(MULTIPLE_REPOSITORIES_SUPPORT_ADDED)) {
+            out.writeStringArray(repositories);
+        } else {
+            out.writeString(repositories[0]);
```

Comment:
This could lead to some strange behaviour in a mixed cluster couldn't it? If I fire a request for multiple snapshots at an older `master` it'll just respond with the snapshots for a single repo. Also, what about special values like `_all` here?

Shouldn't we assert `repositories.length == 1` here and then add logic to ensure we err out on multiple-repository requests in a mixed version cluster instead of quietly returning "something"?

Reply:
I think, `GetSnapshotsResponse.writeTo` method protects us from this problem. https://github.com/elastic/elasticsearch/pull/42090/files#diff-4660cd4d86c46cf0b715f6a338e80034R210
So when running 7.latest and 8.x cluster mixed cluster. If 7.latest node receives the request it will generate `GetSnapshotResponse` and its `writeTo` method will throw `IllegalArgumentException` if snapshots from multiple repos are requested. 
Because of this special values, like "_all", "*" it's hard to tell in advance if snapshots for multiple repos are requested.

Reply:
However, I'm not sure that writeTo/readFrom logic is executed if local transport is used.